### PR TITLE
Spell the C library's log in pinned numpy, and get the design's speed back

### PR DESCRIPTION
**Pull request by jmrplens**
The deterministic weighting design gets most of its speed back without moving a single bit: a cold design now costs about 200 ms where it had come to cost about 260 (and about 340 on this branch's own baseline measurements), and the corpus of designs is byte-identical before and after, verified as one SHA-256 over the coefficient bytes of 133 (curve, rate) pairs.

The cost was the logarithm. Determinism had put every transcendental on the design path onto a `math.*` loop, because numpy's dispatched kernels disagree in the last bit under AVX512, and `log` is the one transcendental inside the Levenberg-Marquardt iteration: once per grid point per factor per step, three quarters of a million calls per design, interpreter overhead on every one.

The new `filters._pinned_log` removes the loop by spelling the C library's own `log` -- the table-driven routine from ARM's optimized-routines that glibc ships as `__ieee754_log` since 2.28 -- in numpy operations IEEE 754 pins exactly: plain add, subtract, multiply, comparisons, integer bit work and table takes. The routine's fused operation `fma(z, invc, -1)` needs no emulation: Dekker's exact product plus Sterbenz's lemma reproduce it in three plain operations. The four multiply-adds the compiler fused in the near-one polynomial are emulated exactly through error-free transformations, themselves verified bit for bit against `math.fma` over eighty million draws. The tables are copied digit for digit from ARM's MIT-licensed `math/log_data.c`, the file glibc carries verbatim; the second table of the C file is not needed, since it exists only for machines without a fast fma.

What the bit claim rests on, measured rather than asserted: every `_scalar_log` input of the whole corpus was captured -- 91 658 333 distinct values across 133 designs -- and the new function returns `math.log`'s own bits on all of them, plus zero mismatches over about a hundred million adversarial draws covering the full exponent range, the near-one window and its edges, and the subnormals. `tests/filters/test_pinned_log.py` pins a deterministic slice of that comparison, including a spy that captures the inputs of a real design and replays each through `math.log`.

Two structural changes carry the rest of the recovery, both order-preserving: each evaluation's logarithms are batched into one call (the function is elementwise, each group's columns are summed over the same elements in the same order, and the group totals join in the order the group tuple fixes), and the normal equations' upper-triangle index pattern is built once per size instead of once per step. The transcendentals outside the iteration stay on the `math` loop, where a design pays them once.

The published figures follow the measurement: about 200 ms per design, about 215 ms against 377 for a first A-weighted minute of 44.1 kHz audio and 220 against 775 for the 468 curve, cached filtering unchanged at about 18 ms, and the 40-by-12 budget comparison remeasured at about 480 ms. Docstrings, both guide twins and the changelog carry the new numbers.

Gates run locally: the full filters suite (745 tests), packaging and architecture tests, ruff over the tree, mypy over src and scripts, and the corpus digest comparison against main.

One consequence surfaced by the macOS jobs, documented in the module and the changelog: `math.log` is whatever `log` the platform's C library ships, and only glibc ships this routine, so the retired elementwise loop was silently returning different designs on macOS than on Linux (Apple's `log` rounds a handful of inputs to the other neighbouring double). The pinned routine returns its own bits everywhere, so the designs are now identical across platforms as well as across CPUs; on a non-glibc platform they moved by that last ulp once, onto the values every other platform already shipped. The bit-for-bit tests against the local `math.log` are gated on a canary that detects the glibc lineage, and a new test holds every platform's libm to within one ulp of the pinned bits.

An adversarial numerical review of the branch then sharpened the claim, and the wording follows it (33f026c88). The corpus statement stands as measured -- every one of the 91 658 333 log inputs of the corpus returns glibc's own bits, and the reviewer independently regenerated 112 designs on both trees to identical hashes -- but beyond the corpus the agreement with glibc is statistical, not structural: the compiled binary carries multiply-adds the compiler fused on its own beyond the source's one explicit fma, and a targeted search of the band just outside the near-one window, where the logarithm is smallest, found about one input in thirty million whose true logarithm falls almost within a half ulp of both neighbours and where the two implementations part by the last bit. The three inputs found are pinned in the one-ulp test; the module, the changelog and this description now bound the claim instead of absolutising it. None of it can reach a design: the pinned routine is the definition on the design path and its own bits are machine-independent. The fma emulation's docstring likewise now says it is exact on its call-site domains rather than correctly rounded in general.

For the record, the corpus digest quoted above is the SHA-256 over `sos.tobytes()` of `_cached_weighting_sos(curve, fs, True)` for the seven curves A, B, C, D, G, AU, 468 at the nineteen rates 100, 125, 200, 250, 400, 500, 903, 1000, 1500, 2000, 8000, 16000, 22050, 32000, 44100, 48000, 88200, 96000 and 192000 Hz, curve fastest, prefixed per pair with `curve@rate:`.

## Summary by Sourcery

Restore weighting-design performance while preserving deterministic coefficients and making results consistent across platforms.

New Features:
- Add a platform-independent pinned logarithm implementation for deterministic weighting design calculations.

Bug Fixes:
- Make weighting designs consistent across operating systems and CPU vectorization paths, while preserving the existing design corpus on glibc platforms.

Enhancements:
- Reduce cold weighting design time from about 260 ms to about 200 ms by vectorizing logarithm evaluation and reusing normal-equation index patterns.
- Batch logarithm calculations during fitting without changing coefficient ordering or numerical results.
- Bound and document the pinned logarithm's agreement with platform libm implementations, including one-ulp portability guarantees.

Documentation:
- Update changelog, weighting documentation, translated guides, and generated reference content with the performance and portability changes.

Tests:
- Add deterministic numerical tests covering pinned logarithm bit patterns, special values, subnormals, fused multiply-add behavior, design-generated inputs, and cross-platform one-ulp agreement.

Chores:
- Include the required licensing notice for the imported logarithm tables and polynomial coefficients.

## Summary by CodeRabbit

* **Performance**
  * Improved weighting-filter design and fitting performance.
  * Reduced fitting costs to approximately 200 ms per design.
  * Reduced first-call estimates to approximately 215 ms for A-weighting and 220 ms for the 468 curve.

* **Consistency**
  * Preserved existing design results while providing consistent calculations across platforms.

* **Documentation**
  * Updated performance figures across weighting-filter guides, reference documentation, changelog, and localized content.

* **Tests**
  * Added coverage for logarithm accuracy, edge cases, special values, and real weighting-design inputs.

### CODE REVIEW

File: src/phonometry/filters/_pinned_log.py [hunk 1]
```diff
@@ -0,0 +1,577 @@
...
+  so ``(p - 1.0) + e`` rounds the very real number the fused operation rounds,
+  once.
+* The four multiply-adds the compiler fused in the near-one polynomial
+  (``x`` within :math:`[1 - 2^{-4}, 1 + 0x1.09p{-4}]`). Those are emulated
+  exactly by :func:`_fused_multiply_add`, the error-free-transformation
+  emulation: split the product, sum with the addend through two exact sums,
+  and the two roundings that remain land on the fused result.
+* The table lookups, which are numpy takes.
+
+The tables and polynomial coefficients are copied digit for digit from ARM's
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply src/phonometry/filters/_pinned_log.py:40]: Adopted in ea959cb3a: Arm Limited's copyright and the full MIT permission notice now travel with the constants, as the license requires for substantial portions, in a comment block directly above the tables. The project LICENSE stays the project's own.

File: src/phonometry/filters/_weighting_design.py [hunk 2]
```diff
@@ -152,7 +155,7 @@
 #: Both are fixed budgets rather than convergence tests, so the routine always
 #: does the same arithmetic, and they are set where the accuracy curve
 #: flattens: measured over this corpus, raising them to 40 by 12 -- 2.4 times
-#: the work, about 610 ms per design instead of 260 -- moves no (curve, rate)
+#: the work, about 480 ms per design instead of 200 -- moves no (curve, rate)
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply src/phonometry/filters/_weighting_design.py]: The two numbers are two different budgets: the sentence measures raising the fixed budget to 40 Lawson rounds of 12 LM steps -- 2.4 times the work of the shipped 25 by 8 -- and 480 ms is what that raised budget costs, against the shipped budget's 200. Reworded in ea959cb3a so the comparison names the shipped budget explicitly.
